File: nj2k/testData/newJ2k/kotlinApiAccess/CorrectNullabilityDetected.java
//file
import kotlinApi.*;

class A {
    int foo(KotlinClass c) {
        return c.getNullableProperty().length()
               + c.getProperty().length()
               + KotlinClass.Companion.getNullableStaticVar()
               + KotlinClass.Companion.getStaticVar()
               + KotlinClass.Companion.nullableStaticFun(1)
               + KotlinClass.Companion.staticFun(1)
               + KotlinApiKt.nullableGlobalFunction("").length()
               + KotlinApiKt.globalFunction("").length();
    }
}

File: nj2k/testData/newJ2k/kotlinApiAccess/CorrectNullabilityDetected.kt
import kotlinApi.KotlinClass
import kotlinApi.KotlinClass.Companion.nullableStaticFun
import kotlinApi.KotlinClass.Companion.nullableStaticVar
import kotlinApi.KotlinClass.Companion.staticFun
import kotlinApi.KotlinClass.Companion.staticVar
import kotlinApi.globalFunction
import kotlinApi.nullableGlobalFunction

internal class A {
    fun foo(c: KotlinClass): Int {
        return (c.nullableProperty!!.length
                + c.property.length
                + nullableStaticVar!!
                + staticVar
                + nullableStaticFun(1)!!
                + staticFun(1)
                + nullableGlobalFunction("")!!.length
                + globalFunction("").length)
    }
}